src/docfx/build/xref/ExternalXrefMapLoader.cs

```diff
@@ -95,12 +99,41 @@ private static void LoadZipFile(Dictionary<string, Lazy<ExternalXrefSpec>> resul
                     var xrefMap = JsonUtility.Deserialize<XrefMapModel>(errors, reader, path);
                     foreach (var spec in xrefMap.References)
                     {
-                        result.TryAdd(spec.Uid, new Lazy<ExternalXrefSpec>(() => spec));
+                        result.TryAdd(spec.Uid, new Lazy<(string?, ExternalXrefSpec)>(() => (xrefMap.RepositoryName, spec)));
                     }
                 }
             }
         }
 
+        private static Dictionary<string, string> GetStringProperties(ReadOnlySpan<byte> content, params string[] propertyNames)
```

Comment:
consider merge this with `GetXrefSpecPositions` so we only scan the file once. Some xrefmap files like .NET combined can be gigabytes.

Reply:
OK